app/pulp/app/viewsets/task.py

```diff
@@ -0,0 +1,30 @@
+from django_filters.rest_framework import filterset
+
+from pulp.app.models import Task, Worker
+from pulp.app.serializers import TaskSerializer, WorkerSerializer
+from pulp.app.viewsets import NamedModelViewSet
+from pulp.app.viewsets.custom_filters import CharInFilter
+
+
+class TaskFilter(filterset.FilterSet):
+    tags = CharInFilter(name='tag__name', lookup_expr='in')
+
+    class Meta:
+        model = Task
+        fields = ['state', 'worker__name', 'tags']
+
+
+class TaskViewSet(NamedModelViewSet):
+    queryset = Task.objects.all()
+    serializer_class = TaskSerializer
+    endpoint_name = 'tasks'
+    filter_class = TaskFilter
+    http_method_names = ['get', 'delete', 'options']
```

Comment:
We used the DELETE method in pulp 2 to cancel a task, but I'm hoping that won't be the case in the v3 API, since it goes against the normally expected behavior of a REST endpoint. And I don't think we actually want clients to delete task records. So given that, I suspect we can remove ``delete`` from this list; does that match your understanding?